src/ripple/app/consensus/RCLCensorshipDetector.h

```diff
@@ -22,78 +22,50 @@
 
 #include <ripple/shamap/SHAMap.h>
 #include <algorithm>
-#include <map>
+#include <utility>
+#include <vector>
 
 namespace ripple {
 
 template <class TxID, class Sequence>
 class RCLCensorshipDetector
 {
 private:
-    std::map<TxID, Sequence> tracker_;
-
-    /** Removes all elements satisfying specific criteria from the tracker
-
-        @param pred A predicate which returns true for tracking entries that
-                    should be removed. The predicate must be callable as:
-                        bool pred(TxID const&, Sequence)
-                    It must return true for entries that should be removed.
-
-        @note This could be replaced with std::erase_if when it becomes part
-              of the standard. For example:
-
-                prune ([](TxID const& id, Sequence seq)
-                    {
-                        return id.isZero() || seq == 314159;
-                    });
-
-              would become:
-
-                std::erase_if(tracker_.begin(), tracker_.end(),
-                    [](auto const& e)
-                    {
-                        return e.first.isZero() || e.second == 314159;
-                    }
-    */
-    template <class Predicate>
-    void prune(Predicate&& pred)
-    {
-        auto t = tracker_.begin();
-
-        while (t != tracker_.end())
-        {
-            if (pred(t->first, t->second))
-                t = tracker_.erase(t);
-            else
-                t = std::next(t);
-        }
-    }
+    std::vector<std::pair<TxID, Sequence>> tracker_;
 
 public:
     RCLCensorshipDetector() = default;
 
     /** Add transactions being proposed for the current consensus round.
 
-        @param seq      The sequence number of the ledger being built.
         @param proposed The set of transactions that we are initially proposing
                         for this round.
     */
-    void propose(
-        Sequence seq,
-        std::vector<TxID> proposed)
+    void propose(std::vector<std::pair<TxID, Sequence>> proposed)
```

Comment:
Just for your consideration...  There are currently two callers of `propose()`, and they both `std::move` from local `std::vector`s.  That's part of how you're increasing efficiency.  But the current signature does not _require_ them to `move` from their local `vector`s.  If they forget to `move` then everything still compiles and runs just fine, but with slightly less efficiency because the `vector` gets copied.

What's your feeling about changing this signature to...
```
void propose(std::vector<std::pair<TxID, Sequence>>&& proposed)
```
so the use of `std::move` is enforced?  Is there a downside to that change that I'm not spotting?  Thanks for your thoughts.